contracts/0.7.x/libraries/Binary/BinaryArbitrable.sol

```diff
@@ -303,23 +252,32 @@ library BinaryArbitrable {
      *  @param _cursor The round from where to start withdrawing.
      *  @param _count The number of rounds to iterate. If set to 0 or a value larger than the number of rounds, iterates until the last round.
      */
-    function withdrawRoundBatch(
+    function batchWithdrawFeesAndRewards(
         ArbitrableStorage storage self, 
         uint256 _itemID, 
         address payable _beneficiary, 
         uint256 _cursor, 
         uint256 _count
     ) internal {
-
         ItemData storage item = self.items[_itemID];
         require(item.status == Status.Resolved, "Dispute not resolved.");
 
         uint256 maxRound = _count == 0 ? item.rounds.length : _cursor + _count;
-        uint256 reward;
         if (maxRound > item.rounds.length)
             maxRound = item.rounds.length;
-        for (uint256 i = _cursor; i < maxRound; i++)
-            reward += _withdrawFeesAndRewards(self, _itemID, _beneficiary, i);
+        uint256 reward;
+        for (uint256 i = _cursor; i < maxRound; i++) {
+            uint256 roundReward = getWithdrawableAmount(self, _itemID, _beneficiary, i);
+            reward += roundReward;
+
+            uint256[3] storage contributionTo = item.rounds[i].contributions[_beneficiary];
+            contributionTo[PARTY_A] = 0;
+            contributionTo[PARTY_B] = 0;
+
+            // The beneficiary doesn't withdraw fees and rewards from a particular ruling but from any. Therefore, the 'ruling' field of Withdrawal is left undefined (0).
+            if (roundReward > 0)
+                emit Withdrawal(_itemID, i, 0, _beneficiary, roundReward);
+        }
 
         _beneficiary.send(reward); // It is the user responsibility to accept ETH.
```

Comment:
Shouldn't we check if the reward is greater than `0` before sending, same way it's done for appeals?

Reply:
I think in this case it's the user's responsability to know if there is a reward to be withdrawn or not. I wouldn't protect them from themselves